lib/timers.js

```diff
@@ -554,67 +557,73 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
+  var argv = timer._argv;
+  var argc = argv ? argv.length : 0;
+  switch (argc) {
+    // fast-path callbacks with 0-3 arguments
+    case 0:
+      return timer._callback();
+    case 1:
+      return timer._callback(argv[0]);
+    case 2:
+      return timer._callback(argv[0], argv[1]);
+    case 3:
+      return timer._callback(argv[0], argv[1], argv[2]);
+    // more then 3 arguments run slower with .apply
+    default:
+      return timer._callback.apply(timer, argv);
+  }
+}
 
-function Immediate() { }
-
-Immediate.prototype.domain = undefined;
-Immediate.prototype._onImmediate = undefined;
-Immediate.prototype._idleNext = undefined;
-Immediate.prototype._idlePrev = undefined;
 
+function createImmediate(callback, args) {
```

Comment:
I found the reason, and it makes no sense, but the good news is that the workaround is
easy, and yes new Immediate is just as fast, and the weird super-slow performance is gone.
(and the weird context switches and rss/heap discrepancy also go away)

If `new Immediate()` stores to `this` a passed-in function, the code runs slow.
If it's not passed in, or if the callback is not stored, the call runs fast.  The args
parameter can be passed and assigned without penalty.  Annotating the created object
is very fast and also incurs no penalty.  (I have not been able to replicate this anomaly
standalone.)

Measuring small runs of tasks with a tool that subtracts the timing loop overhead,
the difference between createImmediate and new Immediate is 7% or so, close enough.
The microbenchmarks use a much much larger dataset (2m and 4m vs 10k) which seems
to favor createImmediate (by 70%), but the smaller dataset is more likely.

Either way, the immediates processing rate is over 250% faster than before this change
(standalone microbenchmark) or 350% (small runs, overhead subtracted).